### DISCUSSION TITLE
Dynamically access template object value

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#174] [Ideas]
I want to create an input component.
```jinja
{% include "partials/input.html" with name="email", required="true" %}
```

Using the dot notation (`schema.email.errors`), it's not possible to dynamically set the field name for the displayed error message. Is there some solution that could dynamically retrieve a property? E.g. `name="email"` then `schema[name].errors`.

**Comment by ellmetha** [comment]
Why not simply modify your partial so that it takes as input your schema field? That way you could do something like:

```html
{% include "partials/input.html" with field=schema.email, label="My input" %}
```

Additionally, you can make use of the `required?` attribute in your snippet to automatically set the required property on the input. For example:

```html
<div class="form-group">
  <label for="paperInputs1">{{ label }}</label>
  <input type="text" {% if field.errored? %}class="background-danger"{% endif %} name="{{ field.id }}" value="{{ field.value }}"{% if field.required? %} required{% endif %}>
  {% for error in field.errors %}<p class="text-danger">{{ error.message }}</p>{% endfor %}
</div>
```

Otherwise, dynamic property lookups using square brackets are not supported in templates presently.
